### ISSUE TITLE
pytorch 模型转换到keras后，识别结果偏差

### ISSUE THREAD

**Issue by gherao** [MVIG-SJTU/AlphaPose#329]
您好，我用keras重写了FastPose的模型，然后将duc_se.pth的权重转换到keras的模型中，使用keras的模型进行提取关键点，发现提取的关键点和原始pytorch的结果有偏差，请问有没有也发生同样问题的前辈？或者遇到过类似问题的前辈？

![keras_output_image](https://user-images.githubusercontent.com/16779844/56882092-1fefee80-6a95-11e9-8017-8951f21d981f.png)
![pytorch_output_image](https://user-images.githubusercontent.com/16779844/56882101-2d0cdd80-6a95-11e9-868f-b9a2cd0c0ec9.png)

**Comment by Fang-Haoshu** [comment]
看起来是pose网络的输入的padding没做好导致的，应该和keras没有什么关系

**Comment by Fang-Haoshu** [comment]
如果可以，你可以把keras版本的做个PR

**Comment by gherao** [comment]
多谢您的回答！

**Comment by gherao** [comment]
根据您的回答，查了一下，确实是padding的问题。
pytorch中卷积中padding指定的padding=3或padding=1后，实际上是填充0.
import torch.nn as nn
nn.Conv2d(planes, planes, kernel_size=3, stride=stride, padding=1, bias=False)

在keras中卷积中的padding参数只有两个值（之前我们使用的是same），“same”和“valid”，same模式不是使用0填充。而valid是不填充。
所以直接使用keras卷积中padding参数设为same是不能与之前pytorch的padding达到一样的效果。

解决方法是，在keras的卷积中 padding指定为valid或不添加padding参数（默认不指定为valid），在卷积之前使用keras的layers.convolutional的ZeroPadding2D进行padding操作。

from keras.layers.convolutional import Conv2D,ZeroPadding2D

x = ZeroPadding2D(padding=(1, 1))(x)
out = Conv2D(filter2, (3, 3), strides=stride, padding='valid', use_bias=False)(x)
